Samples/Insights/cpp-win32/InsightsSample.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "InsightsSample.h"
+
+/*
+* Boolean Event
+* 
+* "data": {
+        "value": 1,
+        "COMMON_WINDOWSAPPSDK_PARAMS": {
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+    }
+
+* TextPayloadEvent
+* 
+* "data": {
+        "value": "This text is the event payload",
+        "COMMON_WINDOWSAPPSDK_PARAMS": {
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+
+* EventWithUserDefinedType 
+* 
+*  "data": {
+        "value": "This is the user-defined type payload",
+        "COMMON_WINDOWSAPPSDK_PARAMS": {
+            "SDKVersion": "0.9",
+            "SDKExperimentationLevel": "",
+            "IsDebugging": false,
+            "AppSessionGuid": "00000D80-0001-0002-B425-C8C4E284D701"
+        }
+    }
+*/
+
+int main()
+{
+    InsightsSample::BooleanTelemetryEvent(true);

```

Comment:
>InsightsSample::BooleanTelemetryEvent(true); [](http://example.com/codeflow?start=4&length=44)

there is no need to register the provider?

Reply:
Nope.

Reply:
wondering why register is needed in OS code, but not here.

Reply:
The macros take care of registration.